intel_extension_for_transformers/backends/neural_engine/graph/application/CMakeLists.txt

```diff
@@ -27,10 +28,88 @@ target_link_libraries(${TARGET} PUBLIC jblas::jblas)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
-add_subdirectory(ChatLLAMA)
-add_subdirectory(ChatGPTNEOX)
-add_subdirectory(ChatMPT)
-add_subdirectory(ChatFALCON)
-add_subdirectory(ChatSTARCODER)
-add_subdirectory(ChatGPTJ)
-add_subdirectory(ChatDOLLY)
+# pybind
+set(TARGET GptjPyBind)
+add_library_w_warning(${TARGET} SHARED pybind_gptj.cpp)
+target_link_libraries(${TARGET} PUBLIC ne_layers common gptj ${CMAKE_THREAD_LIBS_INIT})
+target_compile_features(${TARGET} PRIVATE cxx_std_11)
+set_target_properties(${TARGET} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+set(TARGET pybind_gptj)
+add_executable_w_warning(${TARGET} pybind_gptj.cpp)
+target_link_libraries(${TARGET} PUBLIC ne_layers common gptj ${CMAKE_THREAD_LIBS_INIT})
+target_compile_features(${TARGET} PRIVATE cxx_std_11)
+
+# all models quant
+set(TARGET quant_falcon)
+add_executable_w_warning(${TARGET} quant_model.cpp)
+warning_check(${TARGET})
+target_compile_definitions(${TARGET} PUBLIC -DMODEL_NAME="falcon")
+target_link_libraries(${TARGET} PUBLIC falcon common ${CMAKE_THREAD_LIBS_INIT})
+target_compile_features(${TARGET} PRIVATE cxx_std_11)
+if(TARGET BUILD_INFO)
+  add_dependencies(${TARGET} BUILD_INFO)
+endif()
+
+set(TARGET quant_gptj)
+add_executable_w_warning(${TARGET} quant_model.cpp)
```

Comment:
unify them into one cmake function
```
function(compile_quant SRC TARGET MODEL_NAME)
 add_executable_w_warning(${TARGET} ${SRC})
  warning_check(${TARGET})
  target_compile_definitions(${TARGET} PUBLIC -DMODEL_NAME=${MODEL_NAME})
  target_link_libraries(${TARGET} PUBLIC ${MODEL_NAME} common ${CMAKE_THREAD_LIBS_INIT})
  target_compile_features(${TARGET} PRIVATE cxx_std_11)
  if(TARGET BUILD_INFO)
    add_dependencies(${TARGET} BUILD_INFO)
  endif()
endfunction()

compile_quant("quant_gptj", quant_model.cpp, "gptj")
compile_quant("quant_falcon", quant_model.cpp, "falcon")
compile_quant("quant_gptjneox", quant_model.cpp, "gptjneox")
compile_quant("quant_dolly", quant_model.cpp, "dolly")
compile_quant("quant_llama", quant_model.cpp, "llama")
compile_quant("quant_mpt", quant_model.cpp, "mpt")
compile_quant("quant_starcoder", quant_model.cpp, "starcoder")
```

Reply:
fixed in https://github.com/intel/intel-extension-for-transformers/pull/137/commits/554e99315bf9e2fb355461a0b868841e867e58fb